pkg/util/cloudprovider.go

```diff
@@ -46,7 +46,7 @@ func DetectCloudProvider() {
 
 	for _, cloudDetector := range detectors {
 		if cloudDetector.callback() {
-			inventories.SetAgentMetadata(inventories.CloudProviderMetatadaName, cloudDetector.name)
+			inventories.SetAgentMetadata("cloud_provider", cloudDetector.name)
```

Comment:
nit: The old code was much less error prone as you didn't need to match a hand hard-coded string value to the one from the file.

Reply:
True, but you did need to match an identifier name, and if you didn't use the identifier name and just passed the string (which 50% of the existing uses did) then it was more error prone.  Neither is perfect -- an Enum would be best -- but I think this is more greppable ("where is cloud_provider set?" -> `git grep cloud_provider`)

Reply:
I like the enum idea - I think that would be an improvement. Hard-coded matching strings as-implemented along with a panic if there's mistakes will eventually cause a problem somewhere.

Reply:
The things I don't like about this approach is that we're duplicating the string in multiple places.

I'm not sure I see this as an improvement compared to a basic list of const strings.